### DISCUSSION TITLE
[Module Federation] tree-shaken for shared modules

### DISCUSSION THREAD

**Discussion by fxxjdedd** [webpack/webpack#12708] [Q&A]
webpack.config.js
```js
    new ModuleFederationPlugin({
      exposes: {
        "./Widget": "./src/Widget",
      },
      shared: {
        "lodash-es": {
          import: "lodash-es"
        } 
      },
    })
```

./src/Widget.js
```js
import { isArray } from 'lodash-es'
export default function Widget() {
  return isArray(...)
}
```

Very curious about how webpack handles `lodash-es` tree-shaken here.

**Comment by fxxjdedd** [comment]
I tried it myself and found that lodash-es was packaged into a chunk. There is no tree-shaken.

This may not be a serious problem, but I am curious what the mechanism is here.

![image](https://user-images.githubusercontent.com/13739947/108326904-c0347b80-7205-11eb-8ccd-c823cf3f4cce.png)

**Reply by alexander-akait** [reply]
Because you write `lodash-es`, so other remotes can use any function from `lodash-es`

**Reply by fxxjdedd** [reply]
I understand, but it does generate too much content. Even if it may be used by other remotes, it will most likely not use all APIs.
